### ISSUE TITLE
[BUG]  Navigating to virtualization management in a multi cluster, there is a serious error

### ISSUE THREAD

**Issue by WuJun2016** [harvester/harvester#1551]
**Describe the bug**

![image.png](https://images.zenhubusercontent.com/60345555ec1db310c78aa2b8/3931e5dc-c0dc-4af6-9c5d-32248537d856)

**To Reproduce**
1.  In a multi-cluster
2.  click  "virtualization management"  tab
3.  Show harvesterhci.management.cluster type not found  (Unstable reproduction)

**Expected behavior**

**Support bundle**

**Environment:**
 - Harvester ISO version: 
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by n313893254** [comment]
I can not reproduce this issue, but this error is thrown by the UI. So I add a condition to make sure UI won't fetch the not exist type.

**Comment by sirredbeard** [comment]
Steps to reproduce:

1. Install Rancher Desktop 0.6.1 on Windows 22000.346
2. Enable WSL integration with openSUSE Tumbleweed
3. `helm repo add rancher-latest https://releases.rancher.com/server-charts/latest`
4. `kubectl create namespace cattle-system`
5. `kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.5.1/cert-manager.crds.yaml`
6. `helm repo add jetstack https://charts.jetstack.io`
7. `helm repo update`
8. `helm install cert-manager jetstack/cert-manager  --namespace cert-manager  --create-namespace  --version v1.5.1`
9. `kubectl get pods --namespace cert-manager`
10. `helm install rancher rancher-latest/rancher  --namespace cattle-system  --set hostname=rancher.local  --set bootstrapPassword=admin`
11. `kubectl -n cattle-system rollout status deploy/rancher`
12. `echo "127.0.0.1 rancher.local" | sudo tee -a /etc/hosts` - add rancher.local to WSL hosts file
13. `ip addr show eth0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}'` - get WSL instance IP address
14. `powershell.exe Start-Process notepad.exe 'c:\Windows\System32\Drivers\etc\hosts' -Verb runAs`
15. Add result from step 13 to Windows hosts file as rancher.local, e.g. `172.23.206.8 rancher.local`
16. Open rancher.local in a Windows browser, login, set new password
17. Navigate to Virtualization Management



![image](https://user-images.githubusercontent.com/33820650/142414228-f157701c-e4fe-4968-83dd-dc14950bf654.png)

![image](https://user-images.githubusercontent.com/33820650/142418839-c342d8cb-5a40-4444-81b3-d1496e6b8315.png)

![image](https://user-images.githubusercontent.com/33820650/142420211-bc869d8d-98ac-43e0-9343-62d4e49930b7.png)

![image](https://user-images.githubusercontent.com/33820650/142420497-b3a03ca1-a11e-4745-9169-61110c77916a.png)

![image](https://user-images.githubusercontent.com/33820650/142422124-0deb825b-875b-4a42-b8a3-b481e147bcc5.png)

![image](https://user-images.githubusercontent.com/33820650/142423108-cc8cd65b-51fd-434a-b24f-4c17e26eb54c.png)

![image](https://user-images.githubusercontent.com/33820650/142423966-c39baad6-dfe4-4762-ac59-f68e0e359e93.png)

![image](https://user-images.githubusercontent.com/33820650/142428176-e2bed6c0-8db3-42a6-9a07-6873d8496910.png)

![image](https://user-images.githubusercontent.com/33820650/142428424-b0232b34-9f42-4951-8c02-e77e6f989608.png)

![image](https://user-images.githubusercontent.com/33820650/142428530-78c29204-3ab0-4231-9edf-7b69811f3e13.png)

```
hayden@T730:/mnt/c/Users/hayde> kubectl cluster-info && kubectl get nodes && kubectl get pods --all-namespaces
Kubernetes control plane is running at https://172.23.202.169:6443
CoreDNS is running at https://172.23.202.169:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
Metrics-server is running at https://172.23.202.169:6443/api/v1/namespaces/kube-system/services/https:metrics-server:/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
NAME   STATUS   ROLES                          AGE   VERSION
t730   Ready    builder,control-plane,master   10m   v1.21.3+k3s1
NAMESPACE                   NAME                                      READY   STATUS      RESTARTS   AGE
kube-system                 local-path-provisioner-5ff76fc89d-ndlf4   1/1     Running     0          10m
kube-system                 metrics-server-86cbb8457f-4t9hx           1/1     Running     0          10m
kube-system                 coredns-7448499f4d-7wkpt                  1/1     Running     0          10m
kube-system                 helm-install-traefik-crd-j5mxm            0/1     Completed   0          10m
kube-system                 svclb-traefik-55s49                       2/2     Running     0          8m27s
kube-system                 helm-install-traefik-b66pb                0/1     Completed   1          10m
kube-image                  builder-67ql8                             2/2     Running     0          8m37s
kube-system                 traefik-97b44b794-hjfrn                   1/1     Running     0          8m27s
cert-manager                cert-manager-56b686b465-tppsn             1/1     Running     0          7m10s
cert-manager                cert-manager-cainjector-75c94654d-879pp   1/1     Running     0          7m10s
cert-manager                cert-manager-webhook-69bd5c9d75-c7tl7     1/1     Running     0          7m10s
cattle-system               rancher-996475d64-qqftr                   1/1     Running     1          6m29s
cattle-system               helm-operation-7bj6s                      0/2     Completed   0          5m51s
cattle-fleet-system         gitjob-5778966b7c-nvvzb                   1/1     Running     0          5m40s
cattle-fleet-system         fleet-controller-66cc4c6b5b-r56zt         1/1     Running     0          5m40s
cattle-system               helm-operation-r4lt6                      0/2     Completed   0          5m43s
cattle-fleet-local-system   fleet-agent-59b74595c-tvnfx               1/1     Running     0          5m35s
cattle-system               rancher-webhook-6979fbd4bf-jfnf5          1/1     Running     0          5m34s
cattle-system               helm-operation-mg2rc                      0/2     Completed   0          5m37s
cattle-system               rancher-996475d64-drcxg                   1/1     Running     2          6m29s
cattle-system               rancher-996475d64-247f8                   1/1     Running     0          6m29s
cattle-system               helm-operation-2n5pk                      0/2     Completed   0          2m2s
```

**Comment by n313893254** [comment]
When a user goes to the password setup page(the route is `auth/setup`). UI does not `applyProduct` before `loadSchema`

**Reproduce steps**
- Change "first-login" setting to `true` via API
- Refresh UI
- Login as before
- Navigate to Virtualization Management

**Verify steps**
- Install the latest Rancher
- Login
- Navigate to Virtualization Management

**Comment by sirredbeard** [comment]
Is there a workaround I could apply now to get this working, @n313893254?

**Comment by n313893254** [comment]
@sirredbeard Refresh the UI can be a workaround，the issue only reproduce on user first login

**Comment by TachunLin** [comment]
Verified **fixed** on rancher v2.6-head. Close this issue. 

(Notice: in rancher v2.6.2 and v2.6.3-rc1 still have this issue)

### Result
In rancher `v2.6-head`, first time login and access virtualization management page `will not` display 
serious error and cannot reload page as well.   

![1455](https://user-images.githubusercontent.com/29251855/143514139-dab6c717-b092-4cb2-a0a7-88b1437928a0.gif)

### Test Information
* Rancher version: v2.6.2, v2.6.3-rc1, v2.6-head

### Verify Steps

1. Install rancher v.2.6-head through docker ommand 
`docker run -d --restart=unless-stopped -p 80:80 -p 443:443 --privileged rancher/rancher:v2.6-head`
2. First time login rancher 
3. Open virtualization management page

### Additional Context
Workaround: press ctrl+r to refresh page
![1455_workaround](https://user-images.githubusercontent.com/29251855/143514833-f7b7f218-f272-4cf7-ab9e-72accec99caf.gif)
